src/App.tsx

```diff
@@ -15,33 +14,131 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE = 'none',
+  ALPABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+  minLength: number,
+) {
+  const visibleGoods = [...goods].filter(element => (
+    element.length >= minLength
+  ));
+
+  if (sortType === SortType.ALPABET) {
+    visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
```

Comment:
perhaps, switch case could better fit it? leaving else-if-statement as the last one is not a common practice